packages/mdc-shape/_functions.scss

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright 2018 Google Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions://
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.//
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+@import './variables';
+
+//
+// Flips the radius values based on RTL context.
+//
+// Examples:
+//
+// 1. mdc-shape-flip-radius((0, 4px, 4px, 0)) => 4px 0 0 4px
+// 2. mdc-shape-flip-radius((0, 8px)) => 8px 0
+//
+@function mdc-shape-flip-radius($radius) {
+  @if length($radius) == 4 {
+    @return nth($radius, 2) nth($radius, 1) nth($radius, 4) nth($radius, 3);
+  } @else if length($radius) == 3 {
+    @return nth($radius, 2) nth($radius, 1) nth($radius, 2) nth($radius, 3);
+  } @else if length($radius) == 2 {
+    @return nth($radius, 2) nth($radius, 1);
+  } @else {
+    @return $radius;
+  }
+}
+
+//
+// Resolves the pill radius to appropriate radius value based on component height.
+// Use this for fixed height components only.
+//
+// Examples:
+//
+// mdc-shape-resolve-pill-radius(36px, pill) => `18px` (i.e., 36px / 2)
+//
+@function mdc-shape-resolve-pill-radius($component-height, $radius) {
+  @if type-of($radius) == "list" {
+    $radius-value: ();
+    @each $corner in $radius {
+      $radius-value: append($radius-value, mdc-shape-resolve-pill-corner_($component-height, $corner));
+    }
+    @return $radius-value;
+  } @else {
+    @return mdc-shape-resolve-pill-corner_($component-height, $radius);
+  }
+}
+
+@function mdc-shape-resolve-pill-corner_($component-height, $radius) {
+  @if $radius == pill {
+    @return $component-height / 2;
+  } @else {
+    @return $radius;
+  }
+}
+
+//
+// Returns $radius value of shape category - `large`, `medium` or `small`.
+// Otherwise, it returns the $radius itself if valid.
+// $radius can be a single value or list of 4.
+//
+// Examples:
+//
+// mdc-shape-prop-value(small) => 4px
+//
+@function mdc-shape-prop-value($radius) {
+  @if type-of($radius) == "list" {
+    @if length($radius) > 4 {
+      @error "Invalid radius: '#{$radius}' radius is not supported";
```

Comment:
This error message could be more specific for what it's being raised for

Reply:
Done.